core/capabilities/triggers/logevent/service.go

```diff
@@ -0,0 +1,161 @@
+type Payload struct {
+	// Time that Log Event Trigger's task execution occurred (RFC3339Nano formatted)
+	ActualExecutionTime string
+}
+
+// Log Event Trigger Capabilities Manager
+// Manages different log event triggers using an underlying triggerStore
+type TriggerService struct {
+	capabilities.CapabilityInfo
+	capabilities.Validator[RequestConfig, Input, capabilities.TriggerResponse]
+	lggr           logger.Logger
+	triggers       CapabilitiesStore[logEventTrigger, capabilities.TriggerResponse]
+	relayer        core.Relayer
+	logEventConfig Config
+	stopChan       services.StopChan
+	wg             sync.WaitGroup
+}
+
+// Common capability level config across all workflows
+type Config struct {
+	ChainID        uint64 `json:"chainId"`
+	Network        string `json:"network"`
+	LookbackBlocks uint64 `json:"lookbakBlocks"`
+	PollPeriod     uint32 `json:"pollPeriod"`
+}
+
+func (config Config) Version(capabilityVersion string) string {
+	return fmt.Sprintf(capabilityVersion, config.Network, config.ChainID)
+}
+
+type Params struct {
+	Logger         logger.Logger
+	Relayer        core.Relayer
+	RelayerSet     core.RelayerSet
+	LogEventConfig Config
+}
+
+var _ capabilities.TriggerCapability = (*TriggerService)(nil)
+var _ services.Service = &TriggerService{}
+
+// Creates a new Cron Trigger Service.
+// Scheduling will commence on calling .Start()
+func NewLogEventTriggerService(p Params) *TriggerService {
+	l := logger.Named(p.Logger, "LogEventTriggerCapabilityService")
+
+	logEventStore := NewCapabilitiesStore[logEventTrigger, capabilities.TriggerResponse]()
+
+	s := &TriggerService{
+		lggr:           l,
+		triggers:       logEventStore,
+		relayer:        p.Relayer,
+		logEventConfig: p.LogEventConfig,
+		stopChan:       make(services.StopChan),
+		wg:             sync.WaitGroup{},
+	}
+	s.CapabilityInfo, _ = s.Info(context.Background())
+	s.Validator = capabilities.NewValidator[RequestConfig, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: s.CapabilityInfo})
+	return s
+}
+
+func (s *TriggerService) Info(ctx context.Context) (capabilities.CapabilityInfo, error) {
+	return capabilities.NewCapabilityInfo(
+		s.logEventConfig.Version(ID),
+		capabilities.CapabilityTypeTrigger,
+		"A trigger that listens for specific contract log events and starts a workflow run.",
+	)
+}
+
+// Register a new trigger
+// Can register triggers before the service is actively scheduling
+func (s *TriggerService) RegisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) (<-chan capabilities.TriggerResponse, error) {
+	if req.Config == nil {
+		return nil, errors.New("config is required to register a log event trigger")
+	}
+	reqConfig, err := s.ValidateConfig(req.Config)
+	if err != nil {
+		return nil, err
+	}
+	// Add log event trigger with Contract details to CapabilitiesStore
+	respCh, err := s.triggers.InsertIfNotExists(req.TriggerID, func() (*logEventTrigger, chan capabilities.TriggerResponse, error) {
```

Comment:
You could avoid the linter issues by packaging this up as a method:
```suggestion
	respCh, err := s.triggers.InsertIfNotExists(req.TriggerID, s.gooMethodName(reqConfig))
```

Reply:
Fixed linter issues